service/matching/matching_engine.go

```diff
@@ -323,7 +330,6 @@ func (e *matchingEngineImpl) getTaskQueuePartitionManagerNoWait(
 
 		if !ok {
 			pm.Start()
-			e.updateTaskQueueGauge(pm, false, 1)
```

Comment:
any particular reason to move this call to inside newTaskQueuePartitionManager?

Reply:
sounds good. but then maybe we should put both increment and decrement calls in the Start and Stop methods of taskQueuePartitionManagerImpl and physicalTaskQueueManagerImpl? this way we'll also catch from the metrics if for example Stop is not called on some instances of these structs.

Reply:
Oh, that's a good idea. Making the changes.